### ISSUE TITLE
Allow only passwordless authentication

### ISSUE THREAD

**Issue by mffap** [zitadel/zitadel#8996]
### Preflight Checklist

- [x] I could not find a solution in the existing issues, docs, nor discussions
- [ ] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

I can allow users to login with username/password only, external IdP only, but not only with Passwordless.

### Describe your ideal solution

As a Manager, I want to set a security policy that allows only login with passwordless authenticators. 

Ideally, the configuration will be simplified and more resistant: Enable / Disable authentication methods. Don't allow users to disable the last available authentication method and avoid lock out.

### Environment

ZITADEL Cloud

### Additional Context

related to #8995

**Comment by hifabienne** [comment]
@livio-a @peintnermax is that something we have already resolved with the new session api/typescript login?

**Comment by peintnermax** [comment]
Yes, the new login will support to login with passkeys only
We have also planned loginSetting changes which allow for simpler configuration. see https://github.com/zitadel/zitadel/issues/8656

**Comment by hifabienne** [comment]
Ok, but this means for now the label is wrong, and it should change, when customers switch to the new login.
And should probably be included in the following issue, livio is currently working on: https://github.com/zitadel/zitadel/issues/8702

**Comment by hifabienne** [comment]
Sorry comment should be for this issue: https://github.com/zitadel/zitadel/issues/8995

**Comment by livio-a** [comment]
Discussed with @muhlemmer
We'll move that to the backlog now. The current behavior the backend supports is allow local or federated authentication.
We can handle passwordless only, resp. allow specific authentication mechanisms using User Schema and the mentioned new login policy.

**Comment by ysemennikov** [comment]
Hello everyone,

Are there any updates on this issue? I'm currently working on a project that requires passwordless authentication - preferably via SMS, or at least Email OTP / magic links.

Right now, I'm considering three possible flows:
- Enter phone number → receive OTP by SMS → enter the code → authenticated
- Enter email → receive OTP by email → enter the code → authenticated
- Enter email → receive a magic link → open the link → authenticated

I've evaluated several open-source identity solutions, but none of them seem to fully support these flows:
- Zitadel: unfortunately no passwordless authentication support yet
- Keycloak: too heavy for the project, harder to integrate and maintain, older UI concepts
- Ory Kratos: requires building the entire UI from scratch, which is too time-consuming for a simple login page
- Casdoor: supports all three flows, but I have security concerns; the JS SDK isn’t actively maintained and the overall design feels dated
- Logto: too many limitations in the open-source version, very limited customization

So at the moment, the only option I see is building a custom UI for Zitadel. But again - that feels like unnecessary overhead for this project.

Passwordless authentication is becoming more and more common, and it looks like the existing open-source identity tools still lack comprehensive support for it. Adding proper passwordless support could be a significant advantage for Zitadel.

What do you think?

**Comment by defo10** [comment]
I faced the same issue. Even though ZITADEL was my top pick, we strongly required passwordless auth via mail otp, so we sadly had to resort to other auth solutions.

**Comment by laurentpellegrino** [comment]
Same issue here. We have to forbid password authentication.

**Comment by qussaif10** [comment]
Facing the same issue here.

**Comment by muhlemmer** [comment]
A simple upvote would suffice, please don't populate the thread.

**Comment by muhlemmer** [comment]
@ysemennikov 

> Zitadel: unfortunately no passwordless authentication support yet

Wrong: we fully support [passwordless](https://zitadel.com/docs/concepts/features/passkeys). 
The scope of this issue is adding the ability to configure the out-of-the-box login and disable certain factors like password. As you mentioned yourself, if the default login is not to your liking you can build a custom one.

> So at the moment, the only option I see is building a custom UI for Zitadel. But again - that feels like unnecessary overhead for this project.

Somehow your requirements are very enterprise-y but customizing a login for your own needs is necessary overhead. That sounds like a contradiction. We are open for contributions in the form of a PR or if we get a request from enterprise customers, we will prioritise implementation by Zitadel.

**Comment by monolithed** [comment]
> Wrong: we fully support [passwordless](https://zitadel.com/docs/concepts/features/passkeys).
> The scope of this issue is adding the ability to configure the out-of-the-box login and disable certain factors like password. As you mentioned yourself, if the default login is not to your liking you can build a custom one.

I also don’t get why Zitadel has an archaic password login but no login via one-time codes to email or phone. Supporting those alongside passkeys would make sense. And I still haven’t found a proper workaround that doesn’t require either creating a user first or sending the code manually.

**Comment by elinashoko** [comment]
This is supported in the next iteration through configurable user schemas and authentication flows. Customers can enable only passkeys for a user schema and define passkey-only registration and login flows, with no password step or fallback.

The built-in `passkey-first` preset still includes password as a recovery option, but this is a preset choice rather than a platform limitation. I’ll therefore close this as addressed by the new authentication model. User schemas are now available for testing: https://zitadel.com/blog/next-zitadel-preview-your-user-model-not-ours

**Comment by monolithed** [comment]
@elinashoko I noticed that neither your article nor your comment above mention OTP (SMS or email) as a primary, passwordless sign‑in method. 
In many parts of the world, passkeys (and biometric hardware like MacBooks) are simply not part of the local practice. There's no widespread adoption, no user habit, and often low digital literacy for such flows. SMS and email codes are the de‑facto standard for passwordless login in most regions — they work on any basic phone, require no special hardware, and users already know how to use them.
Are there any concrete plans to add OTP as a standalone first‑factor method in the near future?
